examples/vae_simple.py

```diff
@@ -0,0 +1,322 @@
+    X_test = load_mnist_images('t10k-images-idx3-ubyte.gz')
+    X_train, X_val = X_train[:-10000], X_train[-10000:]
+    return X_train, X_val, X_test
+
+# ############################# Output images ################################
+# image processing using PIL
+
+def get_image_array(X, index, shp=(28,28), channels=1):
+    ret = (X[index] * 255.).reshape(channels,shp[0],shp[1]) \
+            .transpose(2,1,0).clip(0,255).astype(np.uint8)
+    if channels == 1:
+        ret = ret.reshape(shp[1], shp[0])
+    return ret
+
+def get_image_pair(X, Xpr, channels=1, idx=-1):
+    mode = 'RGB' if channels == 3 else 'L'
+    shp=X[0][0].shape
+    i = np.random.randint(X.shape[0]) if idx == -1 else idx
+    orig = Image.fromarray(get_image_array(X, i, shp, channels), mode=mode)
+    ret = Image.new(mode, (orig.size[0], orig.size[1]*2))
+    ret.paste(orig, (0,0))
+    new = Image.fromarray(get_image_array(Xpr, i, shp, channels), mode=mode)
+    ret.paste(new, (0, orig.size[1]))
+    return ret
+
+# ############################# Batch iterator ###############################
+
+def iterate_minibatches(inputs, targets, batchsize, shuffle=False):
+    assert len(inputs) == len(targets)
+    if shuffle:
+        indices = np.arange(len(inputs))
+        np.random.shuffle(indices)
+    for start_idx in range(0, len(inputs) - batchsize + 1, batchsize):
+        if shuffle:
+            excerpt = indices[start_idx:start_idx + batchsize]
+        else:
+            excerpt = slice(start_idx, start_idx + batchsize)
+        yield inputs[excerpt], targets[excerpt]
+
+
+# ##################### Custom layer for middle of VCAE ######################
+# This layer takes the mu and sigma (both DenseLayers) and combines them with
+# a random vector epsilon to sample values for a multivariate Gaussian
+
+class GaussianSampleLayer(MergeLayer):
+    def __init__(self, mu, logsigma, rng=None, **kwargs):
+        self.rng = rng if rng else RandomStreams(nn.random.get_rng().randint(1,2147462579))
+        super(GaussianSampleLayer, self).__init__([mu, logsigma], **kwargs)
+
+    def get_output_shape_for(self, input_shapes):
+        return input_shapes[0]
+
+    def get_output_for(self, inputs, **kwargs):
+        mu, logsigma = inputs
+        shape=(self.input_shapes[0][0] or inputs[0].shape[0],
+                self.input_shapes[0][1] or inputs[0].shape[1])
+        return mu + T.exp(logsigma) * self.rng.normal(shape)
+
+
+# ############################## Build Model #################################
+# encoder has 1 hidden layer, where we get mu and sigma for Z given an inp X
+# continuous decoder has 1 hidden layer, where we get mu and sigma for X given code Z
+# binary decoder has 1 hidden layer, where we calculate p(X=1)
+# once we have (mu, sigma) for Z, we sample L times
+# Then L separate outputs are constructed and the final layer averages them
+
+def build_vae(inputvar, L=1, binary=False, imgshape=(28,28), channels=1, z_dim=2, n_hid=256):
```

Comment:
Any particular reason why you have different defaults here than in `main()`? I think it would be easier to follow if they were all the same!